packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -177,6 +175,20 @@ export const useShippingMethod = props => {
         }
     }, [chosenShippingMethodData]);
 
+    // Simple heuristic to check shipping data existed prior to this render.
+    // On first submission, when we have data, we should tell the checkout page
+    // so that we set the next step correctly.
+    const hasData =
```

Comment:
Nice rename; will do this in scope of my authenticated flow.

Reply:
Yea, I think it's a better indicator of what the condition means. I also want to rename `onSave` to something like.. `onHasData`.

Reply:
`proceedToNextStep`
`incrementStep`
`galavantForward`
`waltzToPayment`

Agreed on the rename, but can happen after this code is more stable.